authors/apps/authentication/serializers.py

```diff
@@ -25,11 +28,78 @@ class Meta:
         # or response, including fields specified explicitly above.
         fields = ['email', 'username', 'password']
 
+    def validate_password(self, password):
+        """ This function validates the password input by a new user signing up
+
+            It ensures that the password length is longer than 8 characters to be considered valid.
+            The password should also contain at least a letter and number. 
+
+        Args: 
+            password (str): This is password string received from user
+
+        Returns: 
+            Returns the validated password
+
+        Raises: 
+            ValidationError: 
+            - ("Password must be longer than 8 characters."): for very short passwords
+            - ("Password should at least contain a number, capital and small letter."): for non alphanumeric paaswords  
+        
+        """
+        if len(password) < 8: 
+            raise serializers.ValidationError("Password must be longer than 8 characters.")
+        elif re.search(r"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]+$", password) is None:
```

Comment:
Just discovered a bug here. a password like this "new.pass.1" without a capital letter passes. 
Try using this regex instead to do proper AND operation on the password string.
'(?=.*[\d])(?=.*[A-Z])(?=.*[a-z])'

Reply:
Noted